codec-http2/src/main/java/io/netty/handler/codec/http2/HttpUtil.java

```diff
@@ -302,7 +303,11 @@ public static Http2Headers toHttp2Headers(FullHttpMessage in) throws Exception {
             HttpResponse response = (HttpResponse) in;
             out.status(new AsciiString(Integer.toString(response.status().code())));
         }
+        return out;
+    }
 
+    public static Http2Headers toHttp2HeadersWithoutReserved(HttpHeaders inHeaders) throws Exception {
```

Comment:
Renamed to toHttp2HeadersWithoutPseudo. "WithoutReserved" was not in reference to the blacklist. It was in reference to not copying reserved/pseudo headers over from the HttpRequest/HttpResponse, because that is not passed as an argument. (And :authority is just a special case.)